tasks/install.yml

```diff
@@ -1,58 +1,50 @@
 ---
-- name: Check if pi-hole is already installed
-  ansible.builtin.stat:
-    path: "/opt/pihole"
-  register: pihole_installed
-  tags:
-    - pihole
-    - installation
-    - check
-
 - name: Create pihole config directory
   become: true
   ansible.builtin.file:
     name: /etc/pihole
     state: directory
-    mode: 0644
-  when: pihole_installed.stat.isdir is undefined
+    mode: "0664"
```

Comment:
`git clone` set repository permission to 0775.
define it to 0664 make task not idempotent

Reply:
During my tests, git clone doesn't change the mode of the main directory. The task was only marked as "changed"  in the first run.
Afterwards, the directory was never changed again.